python/ray/train/tests/test_worker_group.py

```diff
@@ -52,6 +60,11 @@ def test_worker_creation_num_cpus(ray_start_2_cpus):
     wg.shutdown()
 
 
+def test_worker_creation_with_memory(ray_start_2_cpus_and_10kb_memory):
+    wg = WorkerGroup(num_workers=2, additional_resources_per_worker={"memory": 1_000})
+    assert len(wg.workers) == 2
```

Comment:
Can we write a test that creates a virtual cluster with multiple nodes with different amounts of memory resource? Then, assert that the 2 workers get scheduled on the correct nodes?

Ex: A cluster with 1 node with 2KB, 4 nodes with 0.5KB. Make sure we schedule both workers onto the first node.

I want to verify that the memory resource requirement gets passed to core correctly and respected.